src/engine/routing_algorithms/alternative_path_mld.cpp

```diff
@@ -662,7 +662,15 @@ makeCandidateVias(SearchEngineData<Algorithm> &search_engine_data,
     Heap &forward_heap = *search_engine_data.forward_heap_1;
     Heap &reverse_heap = *search_engine_data.reverse_heap_1;
 
+    // All via nodes in the overlapping search space (except the shortest path via node).
+    // Will be filtered and ranked and then used for s,via and via,t alternative paths.
+    std::vector<WeightedViaNode> candidate_vias;
+
     insertNodesInHeaps(forward_heap, reverse_heap, phantom_node_pair);
+    if (forward_heap.Empty() || reverse_heap.Empty())
+    {
+        return candidate_vias;
```

Comment:
Would `return {};` work here? Then you wouldn't need to move the declaration of `candidate_vias`...